moment_kinetics/test/fokker_planck_tests.jl

```diff
@@ -199,27 +190,31 @@ function runtests()
             end
             rpbd_exact = allocate_rosenbluth_potential_boundary_data(vpa,vperp)
             # use known test function to provide exact data
+            begin_s_r_z_anyv_region()
             calculate_rosenbluth_potential_boundary_data_exact!(rpbd_exact,
                   H_M_exact,dHdvpa_M_exact,dHdvperp_M_exact,G_M_exact,
                   dGdvperp_M_exact,d2Gdvperp2_M_exact,
                   d2Gdvperpdvpa_M_exact,d2Gdvpa2_M_exact,vpa,vperp)
             # calculate the potentials numerically
-            calculate_rosenbluth_potentials_via_elliptic_solve!(fkpl_arrays.GG,fkpl_arrays.HH,fkpl_arrays.dHdvpa,fkpl_arrays.dHdvperp,
-                 fkpl_arrays.d2Gdvpa2,fkpl_arrays.dGdvperp,fkpl_arrays.d2Gdvperpdvpa,fkpl_arrays.d2Gdvperp2,F_M,
-                 vpa,vperp,vpa_spectral,vperp_spectral,fkpl_arrays;
-                 algebraic_solve_for_d2Gdvperp2=false,calculate_GG=true,calculate_dGdvperp=true)
+            @views calculate_rosenbluth_potentials_via_elliptic_solve!(
+                 fkpl_arrays.GG[:,:,1], fkpl_arrays.HH[:,:,1], fkpl_arrays.dHdvpa[:,:,1],
+                 fkpl_arrays.dHdvperp[:,:,1], fkpl_arrays.d2Gdvpa2[:,:,1],
+                 fkpl_arrays.dGdvperp[:,:,1], fkpl_arrays.d2Gdvperpdvpa[:,:,1],
+                 fkpl_arrays.d2Gdvperp2[:,:,1], F_M, vpa, vperp, vpa_spectral,
+                 vperp_spectral, fkpl_arrays; algebraic_solve_for_d2Gdvperp2=false,
+                 calculate_GG=true, calculate_dGdvperp=true)
             # extract C[Fs,Fs'] result
             # and Rosenbluth potentials for testing
             begin_vperp_vpa_region()
             @loop_vperp_vpa ivperp ivpa begin
-                G_M_num[ivpa,ivperp] = fkpl_arrays.GG[ivpa,ivperp]
-                H_M_num[ivpa,ivperp] = fkpl_arrays.HH[ivpa,ivperp]
-                dHdvpa_M_num[ivpa,ivperp] = fkpl_arrays.dHdvpa[ivpa,ivperp]
-                dHdvperp_M_num[ivpa,ivperp] = fkpl_arrays.dHdvperp[ivpa,ivperp]
-                dGdvperp_M_num[ivpa,ivperp] = fkpl_arrays.dGdvperp[ivpa,ivperp]
-                d2Gdvperp2_M_num[ivpa,ivperp] = fkpl_arrays.d2Gdvperp2[ivpa,ivperp]
-                d2Gdvpa2_M_num[ivpa,ivperp] = fkpl_arrays.d2Gdvpa2[ivpa,ivperp]
-                d2Gdvperpdvpa_M_num[ivpa,ivperp] = fkpl_arrays.d2Gdvperpdvpa[ivpa,ivperp]
+                G_M_num[ivpa,ivperp] = fkpl_arrays.GG[ivpa,ivperp,1]

```

Comment:
Perhaps I have misunderstood the whole PR -> what does the 1 index? Is it one of the "hidden" anyv buffer arrays for a single ispecies, iz, ir point?